compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -184,7 +184,7 @@ impl<'hir> LoweringContext<'_, 'hir> {
                     )
                 }),
                 ExprKind::TryBlock(body) => self.lower_expr_try_block(body),
-                ExprKind::Match(expr, arms) => hir::ExprKind::Match(
+                ExprKind::Match(expr, arms, _) => hir::ExprKind::Match(
```

Comment:
This should be kept into the HIR, right? Clippy might not handle postfix well, especially on current lints. It should ignore any `MatchKind::Postfix` for AST lints, and `MatchSource::Postfix`.

Most match lints don't care about the scrutinee, which is the only real difference here, but there is `match_on_vec_items` for example which might give a completely wrong suggestion.

Reply:
I do not know much about HIR, but I'll look in to it.

Reply:
Perhaps surprisingly, It Just Works:tm:. Which makes sense, since it doesn't change anything about how the scrutinee & arms are represented in HIR. Then since the span is adjusted for the new scrutinee location, the output is still correct.